### PR TITLE
fix(connectivity_plus):  Do not return ConnectivityResult.none on iOS and MacOS with VPN

### PR DESCRIPTION
**Pull request by mit-73**
## Description

Added support of `ConnectivityResult.vpn` for iOS (12+) and MacOS (10.14+).

## Related Issues

- *Related #857*

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by mit-73** [comment]
@nohli @miquelbeltran ?

**Comment by miquelbeltran** [comment]
sorry, I was going to review and then this fell off my plate, will check now

File: packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift
Comment (miquelbeltran) [inline packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: I am not convinced about this. Of course, "none" is not correct either, but as far as I understand we don't have the certainty that this is a VPN.

Maybe adding another type `other` would be better?

Comment (mit-73) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: I wouldn't want to break the common interface with the android version. What's here, what's in android, vpn doesn't guarantee network connectivity.

It says "other" in the documentation:`The network interface type used for communication over virtual networks or networks of unknown types.`

Anyway, I agree with you that "other" is not "vpn", waiting for a final decision from you.

Comment (miquelbeltran) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: > The network interface type used for communication over virtual networks or networks of unknown types.

Thanks for pointing this out! VPNs do indeed appear as "other", so it kind of makes sense as well.

Any thoughts @nohli @vbuberen ?

Comment (nohli) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: > Anyway, I agree with you that "other" is not "vpn", waiting for a final decision from you.

@miquelbeltran I don't get the last sentence. I would be fine with changing line 26 to `other`.
If it should be added *additionally* to `vpn`, which argument would be passed to `usesInterfaceType()` for `other`?

Comment (mit-73) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: @nohli @miquelbeltran should I change `vpn` to `other` for iOS/MacOS?

Comment (vbuberen) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift:26]: No need to tag people all the time. Those, who are in a thread will get an email anyway.

Yes, agree with suggestion from Miguel to have `other` returned for such cases.

Comment (mit-73) [reply packages/connectivity_plus/connectivity_plus/ios/Classes/PathMonitorConnectivityProvider.swift]: Okay, I changed it to `other`.

File: packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart
Comment (nohli) [inline packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart:23]:
```suggestion
  /// Other: Device is connected to an unknown network
```

File: packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart
Comment (nohli) [inline packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart:20]: Could you write an explanation what this exactly means? In case of vpn it returns other? Only in a simulator, or on any device?

Comment (mit-73) [reply packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart:20]: I don't quite understand how to describe it correctly. Yes in case of vpn it returns other, on any device.

[Documentation describing type other](https://developer.apple.com/documentation/network/nwinterface/interfacetype/other).

Comment (mit-73) [reply packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart:20]: Would that be more correct?

`Note for iOS and macOS: in the case of [vpn] it returns [other], both in the simulator and on any device.`

**Comment by mit-73** [comment]
@nohli I have corrected all your comments.

One test went down with this problem:`Error: Timeout waiting for emulator to boot.`

File: packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart
Comment (nohli) [inline packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart:23]:
```suggestion
  /// There is no separate network interface type for [vpn].
  /// It returns [other] on any device (also simulator).
```

**Review by nohli — APPROVED** [review]
Nice work, thanks for your contribution.
I've added one suggestion for improving the docs.

**Comment by crespinb** [comment]
Hi, when can we expect this to be merged?

**Review by vbuberen — APPROVED** [review]

**Pull request merged**
